File: VehicleServer/Entities/PlatePool.cs
namespace VehicleServer.Entities
{
    public class PlatePool
    {
        public int PlatePoolId { get; set; }
        public int AssignStatus { get; set; }
        public string PlateNumber { get; set; } = string.Empty;
        public int MajorId { get; set; }
        public int? MinorId { get; set; }
        public int PlateSizeId { get; set; }
        public int VehicleCategoryId { get; set; }
        public int PlateRegionId { get; set; }
        public DateTime? CreatedDate { get; set; }
        public string? CreatedByUsername { get; set; }
        public Guid? CreatedByUserId { get; set; }
        public DateTime? LastModifiedDate { get; set; }
        public string? LastModifiedByUsername { get; set; }
        public Guid? LastModifiedByUserId { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletedDate { get; set; }
        public string? DeletedByUsername { get; set; }
        public Guid? DeletedByUserId { get; set; }
        public bool IsActive { get; set; }

        // Navigation properties
        public StockTransaction? StockTransactions { get; set; }
    }
}

File: VechicleClient/src/app/Models/PlatePool.ts
export interface PlatePool {
  platePoolId: number;
  assignStatus: number;
  plateNumber: string;
  majorId: number;
  minorId: number;
  plateSizeId: number;
  vehicleCategoryId: number;
  plateRegionId: number;
  createdDate: string; // You may use Date type if you handle date manipulation
  createdByUsername: string;
  createdByUserId: string;
  lastModifiedDate: string; // You may use Date type if you handle date manipulation
  lastModifiedByUsername: string;
  lastModifiedByUserId: string;
  isDeleted: boolean;
  deletedDate: string | null; // You may use Date type if you handle date manipulation
  deletedByUsername: string | null;
  deletedByUserId: string | null;
  isActive: boolean;
}
